### PR TITLE
android: remove unused code to reduce JSON deserialization overhead

### PR DESCRIPTION
**Pull request by oxtoacart**
Updates tailscale/corp#18202

### CODE REVIEW

**Comment by oxtoacart** [comment]
In my testing this didn't make a huge difference, but if we're not using types/fields, might as well not bother deserializing them.

File: android/src/main/java/com/tailscale/ipn/ui/model/IpnState.kt [hunk 1]
```diff
@@ -5,114 +5,15 @@ package com.tailscale.ipn.ui.model
 
 import kotlinx.serialization.Serializable
 
-class IpnState {
```
Comment (barnstar) [inline android/src/main/java/com/tailscale/ipn/ui/model/IpnState.kt:8]: We don't use this now but there are API endpoints that are yet to be implemented that will require us to fetch PeerStatus (for example). 

Removing this doesn't really gain us much.  Commenting out the unused fields will.

File: android/src/main/java/com/tailscale/ipn/ui/model/NetMap.kt [hunk 2]
```diff
@@ -5,51 +5,49 @@ package com.tailscale.ipn.ui.model
...
-class Netmap {
-    @Serializable
-    data class NetworkMap(
-            var SelfNode: Tailcfg.Node,
-            var NodeKey: KeyNodePublic,
-            var Peers: List<Tailcfg.Node>? = null,
-            var Expiry: Time,
-            var Domain: String,
-            var UserProfiles: Map<String, Tailcfg.UserProfile>,
-            var TKAEnabled: Boolean,
```
Comment (barnstar) [inline android/src/main/java/com/tailscale/ipn/ui/model/NetMap.kt:17]: We'll need some of these fields shortly too.   TAKEnabled before launch.  Domain is used on iOS to disambiguate accounts and tailnet name, etc.

**Review by barnstar — CHANGES_REQUESTED** [review]
We should leave in the (currently) unused types and just comment out the unused fields to reduce the parsing overhead.  A couple of the ones you removed will just be going right back in when we get to tailnet lock and possibly as we refine some of the UI.

**Pull request closed without merging**